frontend/src/components/Header/Header.tsx

```diff
@@ -1,15 +1,84 @@
-import React from "react";
+import React, { useEffect, useState, useRef } from "react";
 import { Link } from "react-router-dom";
 
+import Rank from "../Rank/Rank";
+import Social from "../Social/Social"
+
 interface HeaderProps {
     nextExperimentSlug: string | undefined;
     nextExperimentButtonText: string;
     collectionSlug: string;
     aboutButtonText: string;
     showScore: boolean;
+    totalScore: BigInteger;
+    scoreClass: string;
+    scoreLabel: string;
+    noScoreLabel: string;
 }
 
-export const Header: React.FC<HeaderProps> = ({ nextExperimentSlug, nextExperimentButtonText, collectionSlug, aboutButtonText, showScore }) => {
+export const Header: React.FC<HeaderProps> = ({ nextExperimentSlug, nextExperimentButtonText, collectionSlug, aboutButtonText, showScore, totalScore, scoreClass, scoreLabel, noScoreLabel }) => {
+    
+    const social = {
+        'apps': ['facebook', 'twitter'],
+        'message': `I scored ${totalScore} points`,
+        'url': 'wwww.amsterdammusiclab.nl',
+        'hashtags': ["amsterdammusiclab", "citizenscience"]
+    }
+
+    const useAnimatedScore = (targetScore) => {
+        const [score, setScore] = useState(0);
+
+        const scoreValue = useRef(0);
+
+        useEffect(() => {
+            if (targetScore === 0) {
+                return;
+            }
+
+            let id = -1;
+
+            const nextStep = () => {
+                // Score step
+                const scoreStep = Math.max(
+                    1,
+                    Math.min(10, Math.ceil(Math.abs(scoreValue.current - targetScore) / 10))
+                );
+
+                // Scores are equal, stop
+                if (targetScore === scoreValue.current) {
+                    return;
+                }
+
+                // Add / subtract score
+                scoreValue.current += Math.sign(targetScore - scoreValue.current) * scoreStep;
+                setScore(scoreValue.current);
+
+                id = setTimeout(nextStep, 50);
+            };
+            id = setTimeout(nextStep, 50);
+
+            return () => {
+                window.clearTimeout(id);
+            };
+        }, [targetScore]);
+
+        return score;
+    };
```

Comment:
I think this hook (collection of hooks) is overly complicated. You are setting a timeout in several places, which feels kind of iffy. It's probably also better to use `requestAnimationFrame` than `setTimeout` in this context as it is more efficient and works better with your display refresh rate. Furthermore, I don't think you need the `scoreValue = useRef(0)` hook and you can do without. An example of how it could also be done:

```suggestion
    const useAnimatedScore = (targetScore: number) => {
        const [score, setScore] = useState(0);
    
        useEffect(() => {
            if (targetScore === 0) {
                setScore(0);
                return;
            }
    
            let animationFrameId: number;
    
            const nextStep = () => {
                setScore((prevScore) => {
                    const difference = targetScore - prevScore;
                    const scoreStep = Math.max(1, Math.min(10, Math.ceil(Math.abs(difference) / 10)));
    
                    if (difference === 0) {
                        cancelAnimationFrame(animationFrameId);
                        return prevScore;
                    }
    
                    const newScore = prevScore + Math.sign(difference) * scoreStep;
                    animationFrameId = requestAnimationFrame(nextStep);
                    return newScore;
                });
            };
    
            // Start the animation
            animationFrameId = requestAnimationFrame(nextStep);
    
            // Cleanup function to cancel the animation frame
            return () => {
                cancelAnimationFrame(animationFrameId);
            };
        }, [targetScore]);
    
        return score;
    };
```

Reply:
This was the old toontje hoger code. I commited your changes to it